core/dbt/clients/registry.py

```diff
@@ -31,28 +31,31 @@ def _get(path, registry_base_url=None):
     fire_event(RegistryProgressGETResponse(url=url, resp_code=resp.status_code))
     resp.raise_for_status()
 
+    json_response = (
+        resp.json()  # if this fails, it should raise requests.exceptions.JSONDecodeError and trigger retry
+    )
     # It is unexpected for the content of the response to be None so if it is, raising this error
     # will cause this function to retry (if called within _get_with_retries) and hopefully get
     # a response.  This seems to happen when there's an issue with the Hub.
     # See https://github.com/dbt-labs/dbt-core/issues/4577
-    if resp.json() is None:
+    if json_response is None:
         raise requests.exceptions.ContentDecodingError(
             "Request error: The response is None", response=resp
         )
-    return resp.json()
+    return json_response
+
+
+_get_cached = memoized(_get)
```

Comment:
Is the indenting right here?

Reply:
I think so? Could alternatively use the `@cache` decorator: https://docs.python.org/3/library/functools.html#functools.cache